quick_select/quick_select.py

```diff
@@ -0,0 +1,63 @@
+def partition(a, start, end):
+    """
+    Perform Partition Operation on array a.
+    Time Complexity: o(nLogn)
+    Auxiliary Space: O(n)
+    :param a: Iterable of elements
+    :param start: pivot value for array
+    :param end: right limit of array
+    :return: return i value for function, used in partitioning of array.
+    """
+    i = start - 1
+    pivot = a[end]
+    for j in range(start, end):
+        if a[j] <= pivot:
+            i += 1
+            a[i], a[j] = a[j], a[i]
+    i += 1
+    a[i], a[end] = a[end], a[i]
+    return i
+
+
+def quick_select(a, k):
+    """
+    Perform quick select operation i.e find kth minimum element from array
+    Time Complexity: O(n) for average case and O(n^2) for worst cases
+    param a: Array on which operation would perfom
+    param k: kth minimun element have to find
+    return: returns kth minimum value
+    """
+    if a is None:
+        return None
+    elif len(a) <= k:
+        return None
+    else:
```

Comment:
To pass the codeclimate check what you can do is use a single if statement as
`if a is not None and len(a) <= k:` then the code of `else` inside there, and keep the return None at end, this may pass the check.